prober/http.go

```diff
@@ -403,3 +416,52 @@ func ProbeHTTP(ctx context.Context, target string, module config.Module, registr
 	redirectsGauge.Set(float64(redirects))
 	return
 }
+
+// NewHTTPClientWithLocalAddr wraps NewHTTPClientFromConfig from
+// github.com/prometheus/common/config/http_config.go and
+// adds the ability to specify a local address to bind the client.
+// The difference of this wrapper to the prometheus config
+// version is the specification of a DialContext for the
+// http.RoundTripper. This is currently not possible at a later
+// stage because of many possible RoundTripper wrappings.
+// Adding this to the prometheus config is an alternative
+// to this wrapper.
+func NewHTTPClientWithLocalAddr(cfg *pconfig.HTTPClientConfig, laddr *net.TCPAddr) (*http.Client, error) {
```

Comment:
This is duplicating code with common, it'd be better to extend it. Though that code is just about to get redone.

Reply:
So how should we proceed? Skipping source-address config for http prober and contribute to that common code redoings? Http prober can be made source-address-capable once that work is done.

Reply:
Waiting for that work to complete is probably best.